src/daemon.js

```diff
@@ -7,18 +7,26 @@ const {
   ACTIONS
 } = require('./constants');
 
+const startTime = new Date().getTime();
+
 const DB = require('./dbs'),
       install = require('./install'),
       fatality = require('./fatality');
 
 const newDeployment = deployDoc =>
-  !!deployDoc &&
-  deployDoc._id === HORTI_UPGRADE_DOC &&
-  (deployDoc.action !== ACTIONS.STAGE || !deployDoc.staging_complete);
+      !!deployDoc &&
+      (deployDoc.action !== ACTIONS.STAGE || !deployDoc.staging_complete);
 
-const performDeployment = (deployDoc, mode, firstRun=false) => {
+const performDeployment = (doc, mode, firstRun=false) => {
   let deployAction;
 
+  const getWritableDeployDoc = doc => ({
```

Comment:
I'll start by saying this is one of the hackier parts of the implementation. I'm all ears on how to do this properly -- but here is what I'm targeting. 

Currently, horti writes to the same document that it watches for changes. So horti watches horti_upgrade and then write to horti_upgrade with a bunch of log events and status stuff. My design philosophy is that the satellite should never write to couch - but I'm adapting that here. And saying it can write to "only local docs".  This logic here says "if mode.writeLocalDeployLog is true, write all your status&log stuff into this different document."

In Satellite mode, we watch the design doc for changes. We certainly don't want to write details of the satellite deployment into the ddoc. Thus, we need to watch one document for changes and write to another place once the deployment starts.

So maybe we write to a new doc like 'satellite-status'. But now take the case where there are two instances of the satellite running, and they both notice the change to the ddoc at the same time. Now they conflict and implode.

Really, the upstream service shouldn't care about this satellite's deployment status. So my thought was to just store the logs/status information in a local doc on the satellite itself.